python/mqt/qmap/ph/graph.py

```
@@ -0,0 +1,558 @@
+    Args:
+        r: Pair of beam splitter reflectivities ``[r_in, r_out]``.
+
+    Returns:
+        Fidelity value in ``[0, 1]`` indicating how well the MZI swaps
+        the two input modes.
+    """
+    cache = np.sqrt((1 - r[0]) * r[1]) + np.sqrt(r[0] * (1 - r[1]))
+    return cache**2
+
+
+def generate_beam_splitter_matrix(
```

Ruff (0.15.21)

[warning] Too many branches (13 > 12)

(PLR0912)